File: app/src/main/java/ru/argerd/repo/categoriesOfHelpScreen/CategoriesOfHelpFragment.java
package ru.argerd.repo.categoriesOfHelpScreen;

import android.content.res.Resources;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import ru.argerd.repo.R;

public class CategoriesOfHelpFragment extends Fragment {

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_categories_of_help, null);

        RecyclerView recyclerCategories = view.findViewById(R.id.recycler_categories);
        recyclerCategories.setLayoutManager(new GridLayoutManager(getContext(), 2));

        Toolbar toolbar = getActivity().findViewById(R.id.toolbar);
        toolbar.getMenu().clear();

        TextView toolbarText = getActivity().findViewById(R.id.toolbar_text);
        toolbarText.setText(R.string.help);

        int[] images = {R.drawable.ic_1, R.drawable.ic_2, R.drawable.ic_3, R.drawable.ic_4,
                R.drawable.ic_5};
        String[] nameCategories = {"Дети", "Взрослые", "Пожилые", "Животные", "Мероприятия"};
        recyclerCategories.setAdapter(new AdapterCategories(getContext(), images, nameCategories));

        Resources resources = getResources();
        recyclerCategories.addItemDecoration(new Decorator(
                resources.getDimension(R.dimen.margin_item)));

        return view;
    }
}

File: app/src/main/java/ru/argerd/repo/ui/categoriesOfHelpScreen/CategoriesOfHelpFragment.kt
package ru.argerd.repo.ui.categoriesOfHelpScreen

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ProgressBar
import android.widget.TextView
import androidx.appcompat.widget.Toolbar
import androidx.constraintlayout.widget.Group
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.snackbar.Snackbar
import moxy.MvpAppCompatFragment
import moxy.presenter.InjectPresenter
import ru.argerd.repo.R
import ru.argerd.repo.model.Category
import ru.argerd.repo.presenters.CategoriesPresenter
import ru.argerd.repo.views.CategoriesView

class CategoriesOfHelpFragment : MvpAppCompatFragment(), CategoriesView {
    private lateinit var adapter: AdapterCategories
    private lateinit var recyclerCategories: RecyclerView
    private lateinit var progressBar: ProgressBar
    private lateinit var group: Group

    private lateinit var listOfCategory: List<Category?>

    @InjectPresenter
    lateinit var presenter: CategoriesPresenter

    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_categories_of_help, null)

        recyclerCategories = view.findViewById(R.id.recycler_categories)
        recyclerCategories.layoutManager = GridLayoutManager(context, 2)

        activity?.let {
            val toolbar: Toolbar = activity!!.findViewById(R.id.toolbar)
            val toolbarText = activity!!.findViewById<TextView>(R.id.toolbar_text)

            toolbar.menu.clear()
            toolbarText.setText(R.string.help)
        }

        group = view.findViewById(R.id.categoriesGroup)
        progressBar = view.findViewById(R.id.categoriesProgress)

        adapter = AdapterCategories(context)

        presenter.setListOfCategory()

        val resources = resources
        recyclerCategories.addItemDecoration(Decorator(
                resources.getDimension(R.dimen.margin_item)))

        return view
    }

    override fun hideProgressBar() {
        progressBar.visibility = View.GONE
    }

    override fun showListOfCategories(list: List<Category?>?) {
        listOfCategory = list ?: listOf()
        adapter.setListOfCategories(listOfCategory)
        adapter.notifyDataSetChanged()
        recyclerCategories.adapter = adapter
        group.visibility = View.VISIBLE
        presenter.hideProgressBar()
    }

    override fun showSnackbarWithError(t: Throwable?) {
        view?.let {
            Snackbar.make(it, t.toString(), Snackbar.LENGTH_LONG).show()
        }
    }
}